### PR TITLE
Dropdown: support `.dropdown-item` wrapped in `<li>` tags

### LINKED ISSUE #33625: Issues with Bootstrap 5 dropdown tabs
In this codepen, if you select one dropdown it'll 'activate' the other:

https://codepen.io/cpsievert/pen/xxgWQjp

Screen Shot 2021-04-13 at 3 58 22 PM
 
Also, note that the codepen uses the following markup:

```html
<ul class="dropdown-menu">
      <a class="dropdown-item" href="#">Action</a>
</ul>
```

But, if you use [the markup structure that the docs recommend](https://getbootstrap.com/docs/5.0/components/navs-tabs/#tabs-with-dropdowns), namely:

 ```html
<ul class="dropdown-menu">
      <li><a class="dropdown-item" href="#">Action</a></li>
</ul>
```

Then you'll get other, different, issues with the `.active` class not being updated properly...

### PR DESCRIPTION
**Pull request by cpsievert**
Fully closes #33625 (follow up to #33626)

This adds support for `.dropdown-item`s to be wrapped in `<li>` tags which is [the HTML structure that the docs currently recommend](https://getbootstrap.com/docs/5.0/components/navs-tabs/#tabs-with-dropdowns).

It'd be great to backport this fix to BS4 as well since that suffers from the same problem, which is a pretty unfortunate change from BS3 (which required `<a>` tags to be wrapped in `<li>`)

### CODE REVIEW

**Comment by XhmikosR** [comment]
@cpsievert feel free to backport it to the v4-dev after this one lands :)

**Comment by GeoSot** [comment]
@cpsievert  please rebase your branch

File: js/tests/unit/tab.spec.js [hunk 1]
```diff
@@ -532,6 +532,28 @@ describe('Tab', () => {
...
     })
 
+    it('should support li > .dropdown-item', () => {
+      fixtureEl.innerHTML = [
+        '<ul class="nav nav-tabs">',
+        '  <li class="nav-item"><a class="nav-link active" href="#home" data-bs-toggle="tab">Home</a></li>',
+        '  <li class="nav-item"><a class="nav-link" href="#profile" data-bs-toggle="tab">Profile</a></li>',
+        '  <li class="nav-item dropdown">',
+        '    <a class="nav-link dropdown-toggle" data-bs-toggle="dropdown" href="#">Dropdown</a>',
+        '    <div class="dropdown-menu">',
```
Comment (alpadev) [inline js/tests/unit/tab.spec.js:542]: For the sake of being valid html, can you change that into a `<ul>` element? :)

**Review by alpadev — APPROVED** [review]
IMHO 
https://github.com/twbs/bootstrap/blob/25b03ca08eed5c793726c2b694a979584991004b/js/src/tab.js#L165-L179

Could be a bit optimized like
```js
const context = element.closest(`${SELECTOR_DROPDOWN},${SELECTOR_NAV_LIST_GROUP}`) // this is so we're not reaching out of tabs nested in a dropdown (not sure if this could be case)
// if context matches SELECTOR_DROPDOWN we know we are still in the dropdown
if (context && context.matches(SELECTOR_DROPDOWN)) {
  SelectorEngine.find(SELECTOR_DROPDOWN_TOGGLE, context)
    .forEach(dropdown => dropdown.classList.add(CLASS_NAME_ACTIVE))
  element.setAttribute('aria-expanded', true)
}
```
Also not sure about `aria-expanded` because this would be added to the dropdown item and that isn't expanded? 🤔

But to keep changes as small as possible, we better go with this one, I guess?  @XhmikosR

**Comment by XhmikosR** [comment]
Yup, definitely split the changes into separate PRs so that we can track any breakage :)

**Comment by XhmikosR** [comment]
@alpadev you want to apply your optimization patch here? Or do you plan to make a PR after this one has landed?

**Comment by alpadev** [comment]
@XhmikosR as you prefer. I was thinking to do this after, so we're unlikely to introduce more bugs before stable.

**Comment by XhmikosR** [comment]
Alright yeah let's do it separately to be safe

**Pull request merged**